### PR TITLE
[DataGrid] Refactor `GridFilterItem` props

### LINKED ISSUE #5604: [data grid] Refactor GridFilterItem properties
# Summary

Breaking change for v6. 
For more context: #3287 

- [ ] Rename `GridFilterItem['columnField']` -> `GridFilterItem['field']` 
  https://github.com/mui/mui-x/blob/b778028da23e7455c985c93c53366ab63848ee6d/packages/grid/x-data-grid/src/models/gridFilterItem.ts#L13

- [ ] Rename `GridFilterItem['operatorValue']` -> `GridFilterItem['operator']`  
  https://github.com/mui/mui-x/blob/b778028da23e7455c985c93c53366ab63848ee6d/packages/grid/x-data-grid/src/models/gridFilterItem.ts#L23

  Context: https://github.com/mui/mui-x/pull/1722#discussion_r640583455

- [ ] Make  `GridFilterItem['operatorValue']` (renamed 'operator') required 

  Context: https://github.com/mui/mui-x/pull/3671#discussion_r804695140
see this comment](https://github.com/mui/mui-x/pull/3671#discussion_r804695140)

# Motivation

### PR DESCRIPTION
**Pull request by MBilalShafi**
Closes #5604 

- [x] Rename GridFilterItem['columnField'] -> GridFilterItem['field']
- [x] Rename `GridFilterItem['operatorValue']` -> `GridFilterItem['operator']`  
- [x] Make  `GridFilterItem['operatorValue']` (renamed 'operator') required 
- [x] Update migration guide

### CODE REVIEW

File: packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx [hunk 1]
```diff
@@ -180,17 +180,27 @@ export const useGridFilter = (
...
 
+        const targetColumn = apiRef.current.getColumn(targetColumnField);
+
         if (filterItemOnTarget) {
           newFilterItems = filterItemsWithValue;
         } else if (props.disableMultipleColumnsFiltering) {
-          newFilterItems = [cleanFilterItem({ columnField: targetColumnField }, apiRef)];
+          newFilterItems = [
+            cleanFilterItem(
+              { field: targetColumnField, operator: targetColumn!.filterOperators![0].value! },
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx:193]: @flaviendelangle Based on this [conversation](https://github.com/mui/mui-x/pull/3671#discussion_r804695140), do you think this approach (same used in `cleanFilterItem`) should be fine for generating `operator` for new filters?

File: docs/data/data-grid/filtering/filtering.md [hunk 2]
```diff
@@ -53,9 +53,9 @@ The filter model is composed of a list of `items` and a `linkOperator`:
 
 A filter item represents a filtering rule and is composed of several elements:
 
-- `filterItem.columnField`: the field on which the rule applies.
+- `filterItem.field`: the field on which the rule applies.
 - `filterItem.value`: the value to look for.
-- `filterItem.operatorValue`: name of the operator method to use (e.g. _contains_), matches the `value` key of the operator object.
+- `filterItem.operator`: name of the operator method to use (e.g. _contains_), matches the `value` key of the operator object.
 - `filterItem.id` ([<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)): only useful when multiple filters are used.
```
Comment (alexfauquette) [inline docs/data/data-grid/filtering/filtering.md:59]: Not directly related to this PR
```suggestion
- `filterItem.id` ([<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)): mandatory when multiple filter Items are used.
```

**Review by alexfauquette — APPROVED** [review]
Nice job 👍 Tha filter API sounds less verbose like that

File: packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx (hunk 1 above)
Comment (MBilalShafi) [reply packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx:193]: > cleanFilterItem can be removed in a follow up.

It's also generating IDs for filter items apart from adding default operator.

https://github.com/mui/mui-x/blob/21553f7d07906ef2d5e553f7437bf3d60ab3ddc6/packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts#L43-L45

Should we rename the function and only handle ID generation part?

**Comment by seemX17** [comment]
This breaking change is making it difficult to restore the state, can we avoid such baseless renames ?
If there is a table state  saved in the DB, reformatting of data is required in order to restore the table.

**Comment by oliviertassinari** [comment]
> If there is a table state saved in the DB, reformatting of data is required in order to restore the table.

Right, it sounds like this would have been better with a backward compatibility logic (a deprecation of the old keys).

**Pull request merged**